src/battle_main.c

```diff
@@ -4004,7 +4004,8 @@ static void HandleTurnActionSelectionState(void)
                     }
                     else if (WILD_DOUBLE_BATTLE
                              && position == B_POSITION_PLAYER_RIGHT
-                             && (gBattleStruct->throwingPokeBall || gChosenActionByBattler[GetBattlerAtPosition(B_POSITION_PLAYER_LEFT)] == B_ACTION_RUN))
+                             && (gBattleStruct->throwingPokeBall || gChosenActionByBattler[GetBattlerAtPosition(B_POSITION_PLAYER_LEFT)] == B_ACTION_RUN)
+                             && gChosenActionByBattler[GetBattlerAtPosition(B_POSITION_PLAYER_LEFT)] != B_ACTION_NOTHING_FAINTED)
```

Comment:
Can you explain why this change is necessary? I've never looked at this part of the code before so I am probably interpreting it wrong, but the original code seems to be implementing the logic "if the left battler is throwing a ball or running, skip the right battler's action". I'm not sure how the left battler could be doing either of those things if it is set to `B_ACTION_NOTHING_FAINTED`, or in other words, I think the addition might be logically equivalent to `&& TRUE`.